numpy/core/src/multiarray/descriptor.c

```diff
@@ -1651,71 +1565,186 @@ PyArray_DescrConverter(PyObject *obj, PyArray_Descr **at)
             *at = _arraydescr_from_ctypes_type(Py_TYPE(obj));
             return *at ? NPY_SUCCEED : NPY_FAIL;
         }
+        _report_generic_error();
+        return NPY_FAIL;
+    }
+}
+
+static int
+_convert_from_bytes(PyObject *obj, PyArray_Descr **at)
+{
+    int check_num = NPY_NOTYPE + 10;
+    int elsize = 0;
+    char endian = '=';
```

Comment:
Dunno if its actually not worse, but `elsize` does not seem to need initialization (I think) and `endian` could also be a `default:` clause (editor always complains about missing default, but not sure I agree with it.

Reply:
This is cleaned up in a future commit.

Reply:
Wait, no it isn't - pushed the change as an update. I've left them with initialization, but put the initialization right next to the use.